superpoint/evaluations/detector_evaluation.py

```diff
@@ -0,0 +1,222 @@
+        warped_col1 = np.add(np.sum(np.multiply(keypoints, H[1, :2]), axis=1), H[1, 2])
+        warped_col2 = np.add(np.sum(np.multiply(keypoints, H[2, :2]), axis=1), H[2, 2])
+        warped_col0 = np.divide(warped_col0, warped_col2)
+        warped_col1 = np.divide(warped_col1, warped_col2)
+        new_keypoints = np.concatenate([warped_col0[:, None], warped_col1[:, None]],
+                                       axis=1)
+        return new_keypoints
+
+    def filter_keypoints(points, shape):
+        """ Keep only the points whose coordinates are
+        inside the dimensions of shape. """
+        mask = (points[:, 0] >= 0) & (points[:, 0] < shape[0]) &\
+               (points[:, 1] >= 0) & (points[:, 1] < shape[1])
+        return points[mask, :]
+
+    def keep_true_keypoints(points, H, shape):
+        """ Keep only the points whose warped coordinates by H
+        are still inside shape. """
+        warped_points = warp_keypoints(points[:, [1, 0]], H)
+        warped_points[:, [0, 1]] = warped_points[:, [1, 0]]
+        mask = (warped_points[:, 0] >= 0) & (warped_points[:, 0] < shape[0]) &\
+               (warped_points[:, 1] >= 0) & (warped_points[:, 1] < shape[1])
+        return points[mask, :]
+
+    paths = get_paths(exper_name)
+    repeatability = []
+    for path in paths:
+        data = np.load(path)
+        shape = data['warped_prob'].shape
+        H = data['homography']
+
+        # Filter out predictions
+        keypoints = np.where(data['prob'] > prob_thresh)
+        keypoints = np.stack([keypoints[0], keypoints[1]], axis=-1)
+        warped_keypoints = np.where(data['warped_prob'] > prob_thresh)
+        warped_keypoints = np.stack([warped_keypoints[0], warped_keypoints[1]], axis=-1)
+        warped_keypoints = keep_true_keypoints(warped_keypoints, np.linalg.inv(H),
+                                               data['prob'].shape)
+
+        # Warp the original keypoints with the true homography
+        true_warped_keypoints = warp_keypoints(keypoints[:, [1, 0]], H)
+        true_warped_keypoints[:, [0, 1]] = true_warped_keypoints[:, [1, 0]]
+        true_warped_keypoints = filter_keypoints(true_warped_keypoints, shape)
+
+        # Compute the repeatability
+        N1 = true_warped_keypoints.shape[0]
+        N2 = warped_keypoints.shape[0]
+        tile1 = np.tile(np.expand_dims(true_warped_keypoints, 1), (1, N2, 1))
+        tile2 = np.tile(warped_keypoints, (N1, 1, 1))
+        # both shapes are now N1 x N2 x 2
+        norm = np.linalg.norm(tile1 - tile2, ord=None, axis=2)
+        count1 = 0
+        count2 = 0
+        if N2 != 0:
+            min1 = np.min(norm, axis=1)
+            count1 = np.sum(min1 <= distance_thresh)
+        if N1 != 0:
+            min2 = np.min(norm, axis=0)
+            count2 = np.sum(min2 <= distance_thresh)
+        if N1 + N2 > 0:
+            repeatability.append((count1 + count2) / (N1 + N2))
+        else:
+            repeatability.append(1)
```

Comment:
I don't think that pairs with no common points should be taken into account at all, consider not appending their result.